src/main.rs

```diff
@@ -18,6 +18,7 @@ use std::time::Instant;
 
 extern crate blas;
 extern crate intel_mkl_src;
+extern crate half;
```

Comment:
my bad! I was looking for "half" instead of "f16". do you think it will perform badly if we use u16 instead? - greater loss of accuracy for sure... just the bucket index, but won't we have to do it that way when we switch to 8 bit quantization (with a lookup table)?

Reply:
on the upside, with f16, for much of the range (lower values...) you will be able to get a more accurate representation, but for the far end of the range most buckets will be unrepresented, so the precision will suffer.
probably f16 is still better overall (but worth testing maybe)